File: Java/AddTwoNumbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode addTwoNumbers(ListNode list1, ListNode list2) {
       ListNode fakeHead = new ListNode(0);
       ListNode curr = fakeHead;
       
       int carry = 0;
       while(list1 != null || list2 != null){
           int value = 0;
           if (list1 != null){
               value += list1.val;
               list1 = list1.next;
           }
           if(list2 != null){
               value += list2.val;
               list2 = list2.next;
           }
           value += carry;
           carry = value / 10;
           
           ListNode next = new ListNode(value % 10);
           curr.next = next;
           curr = next;
       }
       
       if(carry == 1 ){
           ListNode lastOne = new ListNode(1);
           curr.next = lastOne;
       }
       return fakeHead.next;
    }
}

File: Go/Add Two Numbers.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
    newHead := ListNode{}
    curr := &newHead
    
    val := 0
    for l1 != nil || l2 != nil || val > 0{
        if l1 != nil {
            val += l1.Val
            l1 = l1.Next
        }
        
        if l2 != nil {
            val += l2.Val
            l2 = l2.Next
        }
        
        curr.Next = &ListNode{}
        curr = curr.Next
        curr.Val = val%10
        val /= 10
    }
    
    return newHead.Next
}
